utils/utils.js

```diff
@@ -0,0 +1,15 @@
+import Conversation from "../components/conversations/conversation.model.js";
+
+export const addMsgToConversation = async (userId, placeId, msg, isOpened) => {
```

Comment:
u can do all this logic in one mongoose query , please read about findOneAndUpdate and push operator in mongoose , this is the links docs :
1-(https://mongoosejs.com/docs/tutorials/findoneandupdate.html)
2-(https://www.mongodb.com/docs/manual/reference/operator/update/push/)

feel free to search in other resources  or ask chat gpt :)

Reply:
the thing with that method is when there is no document he does not update anything, not returning undefined or some data for us to know that we need to create a new conversation, right?